common/src/main/java/com/hedera/block/common/utils/FileUtilities.java

```diff
@@ -111,22 +109,20 @@ public static void createPathIfNotExists(
     public static void createPathIfNotExists(
             @NonNull final Path toCreate,
             @NonNull final System.Logger.Level logLevel,
-            @NonNull final FileAttribute<Set<PosixFilePermission>> filePermissions,
-            @NonNull final FileAttribute<Set<PosixFilePermission>> folderPermissions,
+            @NonNull final FileAttribute<Set<PosixFilePermission>> permissions,
```

Comment:
This is wrong.  Files and folders MUST NOT have the same permissions.

Reply:
I seem to be confused about this.

Technically we will always either create a directory or a file. We will never need both `filePermissions` and `folderPermissions`, we technically need only one set of permissions and to have the boolean to either create a directory or a folder using the permissions we pass. To me, if I will supply file and folder permissions and the boolean `createDir` is true, I will always use only the `folderPermissions` and if the boolean `createDir` is false, then I will always use only the `filePermissions`, so if I will always use only one set of permissions, why should I be supplying both?

I do understand of course that files and directories must not have the same permissions, that is obvious. But why I must be supplying both, when only one will ever be used based on the boolean? What is the design flaw in supplying only one set of permissions? Is it because it is error prone in the sense that I can supply mistakenly directory permissions and have the boolean `createDir` set to false and vice versa? If that is the reason, then I would argue that forcing the caller to supply both file and folder permissions is the same in terms of being error prone, if not even more error prone since we need to now consider to not mistakenly put the file permissions in the place of the folder permissions as well?

But I believe I am missing something here and I fail to understand the design flaw of having to supply only one set of permissions and having the boolean instead of both permissions and having the boolean as well.

Reply:
Noted, thanks for the explanation, looks like I was on the right track as far as the reasoning behind the design flaw. Since it seems that currently the only usage for the create path utility method is to create directories, I removed support for the file path creation functionality, as it really is the best in terms of design. In the future if it is needed, we could always add a new method that will do the creation of file paths.